sim-cli/src/parsing.rs

```diff
@@ -362,3 +343,37 @@ fn mkdir(dir: PathBuf) -> anyhow::Result<PathBuf> {
     fs::create_dir_all(&dir)?;
     Ok(dir)
 }
+
+pub async fn parse_sim_params(cli: &Cli) -> anyhow::Result<SimParams> {
+    let sim_path = read_sim_path(cli.data_dir.clone(), cli.sim_file.clone()).await?;
+    let sim_params = serde_json::from_str(&std::fs::read_to_string(sim_path)?).map_err(|e| {
+        anyhow!(
+            "Could not deserialize node connection data or activity description from simulation file (line {}, col {}, err: {}).",
+            e.line(),
+            e.column(),
+            e.to_string()
+        )
+        })?;
+    Ok(sim_params)
+}
+
+pub async fn get_validated_activities(
+    clients: &HashMap<PublicKey, Arc<Mutex<dyn LightningNode>>>,
+    nodes_info: HashMap<PublicKey, NodeInfo>,
+    activity: Vec<ActivityParser>,
+) -> Result<Vec<ActivityDefinition>, LightningError> {
+    // We need to be able to look up destination nodes in the graph, because we allow defined activities to send to
+    // nodes that we do not control. To do this, we can just grab the first node in our map and perform the lookup.
+    let get_node = async |pk: &PublicKey| -> Result<NodeInfo, LightningError> {
+        if let Some(c) = clients.values().next() {
+            return c.lock().await.get_node_info(pk).await;
+        }
+        Err(LightningError::GetNodeInfoError(
+            "no nodes for query".to_string(),
+        ))
+    };
+    let (pk_node_map, alias_node_map) = add_node_to_maps(&nodes_info).await?;
+    let validated_activities =
+        validate_activities(activity.to_vec(), pk_node_map, alias_node_map, get_node).await?;
+    Ok(validated_activities)
```

Comment:
nit: can inline to `Ok(validate_activities(...))`, intermediate var doesn't add much